src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp

```diff
@@ -513,49 +582,14 @@ bool TypeLookupManager::create_endpoints()
     else
     {
         EPROSIMA_LOG_ERROR(TYPELOOKUP_SERVICE, "Typelookup reply writer creation failed.");
-        delete builtin_reply_writer_history_;
-        builtin_reply_writer_history_ = nullptr;
-        return false;
+        ret = false;
     }
 
-    ReaderAttributes ratt;
-    ratt.endpoint.unicastLocatorList = builtin_protocols_->m_metatrafficUnicastLocatorList;
-    ratt.endpoint.multicastLocatorList = builtin_protocols_->m_metatrafficMulticastLocatorList;
-    ratt.endpoint.external_unicast_locators = builtin_protocols_->m_att.metatraffic_external_unicast_locators;
-    ratt.endpoint.ignore_non_matching_locators = pattr.ignore_non_matching_locators;
-    ratt.endpoint.remoteLocatorList = builtin_protocols_->m_initialPeersList;
-    ratt.matched_writers_allocation = pattr.allocation.participants;
-    ratt.expectsInlineQos = true;
-    ratt.endpoint.topicKind = fastrtps::rtps::NO_KEY;
-    ratt.endpoint.reliabilityKind = fastrtps::rtps::RELIABLE;
-    ratt.endpoint.durabilityKind = fastrtps::rtps::VOLATILE;
-
-    // Built-in request reader
-    builtin_request_reader_history_ = new ReaderHistory(hatt);
-
-    RTPSReader* req_reader;
-    if (participant_->createReader(
-                &req_reader,
-                ratt,
-                builtin_request_reader_history_,
-                request_listener_,
-                fastrtps::rtps::c_EntityId_TypeLookup_request_reader,
-                true))
-    {
-        builtin_request_reader_ = dynamic_cast<StatefulReader*>(req_reader);
-        EPROSIMA_LOG_INFO(TYPELOOKUP_SERVICE, "Builtin Typelookup request reader created.");
-    }
-    else
+    // Built-in reply reader
+    if (nullptr == reply_listener_)
     {
-        EPROSIMA_LOG_ERROR(TYPELOOKUP_SERVICE, "Typelookup request reader creation failed.");
-        delete builtin_request_reader_history_;
-        builtin_request_reader_history_ = nullptr;
-        delete request_listener_;
-        request_listener_ = nullptr;
-        return false;
+        reply_listener_ = new TypeLookupReplyListener(this);
```

Comment:
Same here, this check is obsolete

Reply:
Removed in [3466f2d](https://github.com/eProsima/Fast-DDS/pull/4240/commits/3466f2db886ab8cffed18814379f228a5b71975e)